xtokens/src/lib.rs

```diff
@@ -706,6 +706,14 @@ pub mod module {
 				}
 			}
 
+			let mut reanchored_recipient = recipient.clone();
+			if recipient == MultiLocation::here() {
+				let ancestry = T::LocationInverter::ancestry();
+				let _ = reanchored_recipient
+					.reanchor(&dest, &ancestry)
+					.map_err(|_| DispatchError::Other("Cannot reanchor recipient"));
```

Comment:
the error type should be `Error::<T>::CannotReanchor`.

Reply:
Also can you put this logic into `do_transfer_multiassets` before invoke `execute_and_send_reserve_kind_xcm`. I think this is a special process only for transfer asset back to statemine/t.

Reply:
Also modified last two paragraph in https://github.com/open-web3-stack/open-runtime-module-library/blob/master/xtokens/README.md

Reply:
Thanks for the review, please take a look at the resolutions.